ff/src/biginteger/macros.rs

```diff
@@ -17,7 +17,16 @@ macro_rules! bigint_impl {
                 let mut carry = 0;
 
                 for i in 0..$num_limbs {
-                    self.0[i] = adc!(self.0[i], other.0[i], &mut carry);
+                    #[cfg(all(target_arch = "x86_64", feature = "asm"))]
```

Comment:
Note that this is a weaker requirement than for the assembly backend; it works on stable, and on all `x86_64` processors, as long as the `asm` feature is enabled. You don't need to have the `bmi2` and `adx` target features enabled.

This means that even stable code can take advantage of faster additions, subtractions, and negations.

Reply:
I think you can change the addcarryx to addcarry with no perf penalty. Addcarryx I think technically needs bmi and adx too

Reply:
Realistically though most x64 processors have both, so I've always wondered whether checking for them is a bit pedantic... But maybe it's fine to be a bit on the safe side. It's just very verbose to compile the asm version, I noticed that for many libraries, the asm versions are the defaults...

Reply:
> I think you can change the addcarryx to addcarry with no perf penalty.

I'll double check this